nifi-nar-bundles/nifi-extension-utils/nifi-record-utils/nifi-avro-record-utils/src/main/java/org/apache/nifi/avro/AvroTypeUtil.java

```diff
@@ -691,6 +697,10 @@ private static boolean isCompatibleDataType(final Object value, final DataType d
                     return true;
                 }
                 break;
+            case MAP:
+                if (value instanceof Map) {
```

Comment:
What kind of flow (which processors, e.g.) did you use to test this with? When I use ConvertRecord, value is a MapRecord not a Map, which causes this not to work. Perhaps we should check for both here?

Reply:
The MapRecord was my initial post, you suggested to open to Map , type. 
the probleme was when the map is present in the union ("null", "record").

Reply:
in debugging initially, the given value was MapRecord,